In [3]:
from IPython.display import Markdown, display
display(Markdown(open("./SM_header.md", "r").read()))

Copyright © 2025 Université Paris Cité

Author: [Guillaume Rousseau](https://www.linkedin.com/in/grouss/), Department of Physics, Paris, France (email: guillaume.rousseau@u-paris.fr)

This archive contains the supplemental materials and replication package associated with the preprint, "*Temporal and topological partitioning in real-world growing networks for scale-free properties study*", available on [arXiv](https://arxiv.org/abs/2501.10145) and [ssrn](http://ssrn.com/abstract=5191689).

**The latest version of the preprint (timestamped arXiv:2501.10145v2) is downloadable here https://arxiv.org/pdf/2501.10145v2**

The current version of the Python scripts and associated resources is available on the [author's GitHub page](https://github.com/grouss/growing-network-study).

This work is currently licensed under the [Creative Commons CC BY-NC-SA 4.0 license](https://creativecommons.org/licenses/by-nc-sa/4.0).

To give appropriate credit and cite this work ([BibTeX entry](./rousseau2025temporal)):
Rousseau, G. (2025). *Temporal and topological partitioning in real-world growing networks for scale-free properties study* [Preprint]. arXiv:2501.10145. https://arxiv.org/abs/2501.10145; also available on SSRN: http://ssrn.com/abstract=5191689

 
# A) Replication Packages

[Open the Replication Package notebook related to the datasets.](./Replication_Package_Datasets.ipynb)

[Open the Replication Package notebook related to the figures.](./Replication_Package_Figures.ipynb)

# B) QuickStart Guide

[Open the QuickStart Guide notebook](./SM00_QuickStart.ipynb)

# C) Table of Contents

- 1. [Function Definitions](./SM01_Functions.ipynb)
- 2. [Dataset Import](./SM02_DatasetImport.ipynb)
- 3. [Building the Transposed Graph](./SM03_BuildingTransposedGraph.ipynb)
- 4. [Temporal Information Quality and Summary Statistics](./SM04_TemporalInformationMainStats.ipynb)
- 5. [Growth Relationship Between Nodes and Edges](./SM05_GrowingRules.ipynb)
- 6. [Topological Partitioning($RV$ Nodes)](./SM06_TopologicalPartitioning.ipynb)
- 7. [In-Degree and Out-Degree Distributions Over Time](./SM07_DegreeDistributionOverTime.ipynb)
- 8. [Distribution Tail Analysis](./SM08_DistributionTailAnalysis.ipynb)
- 9. [Temporal Partitioning](./SM09_TemporalPartitioning.ipynb)
- 10. [Derived $O-(RV/RL)-O$ Graph Construction](./SM10_DerivedGrowingNetwork.ipynb)
- 11. [Building the $TSL$ Partitioning](./SM11_TSLPartitioning.ipynb)
- 12. [Barabási–Albert Model Use Case](./SM12_BarabasiAlbertUseCase.ipynb)


**NB :** As of 2025/05/16, the QuickStart guide, the replication packages, and SM01 to SM12 are available. The Python scripts are also provided under `local_utils` directory, but they are not in their final form and should be considered an alpha release. The graph datasets used in the study are available in a distinct Zenodo Deposit 10.5281/zenodo.15260640 ($\sim50$ Go), including the main dataset $O/RV/RL-O/RV/RL$ (2+ billions of nodes, $\sim4$ billions of edges), and two derived $O-(RV/RL)-O$ graphs ($\sim150$ millions nodes and edges). 

More release notes are available in the [dedicated notebook](./SM_ReleaseNote.ipynb).

# Release Note

This notebook includes important information about the versioning of the preprints and supplemental materials.

## a) Preprints

### V2.0 - 2025-06-09
**Release of version 2.0 of the preprint**

- Includes two new sections (with new figures) dedicated to the comparison with minimal models.  
- Revised all other sections to better highlight the study’s main findings and clarify the connection between the research questions and the analyses performed.

---

### V1.1 - 2025-03-24

Release of version 1.1 of the preprint:  
*Temporal and Topological Partitioning in Real-World Growing Networks for Scale-Free Properties Study*  
([SSRN:5191689](https://papers.ssrn.com/sol3/papers.cfm?abstract_id=5191689)).

This version includes minor revisions, including the splitting of Figure 4 and a switch from letter format to article style.

---

### V1.0 - 2025-01-17

Initial public release of the preprint:  *Temporal and topological partitioning in real-world growing networks for scale-free properties study*  ([arXiv:2501.10145](https://arxiv.org/abs/2501.10145)).

Supplemental materials, datasets, and replication packages are available in a dedicated Zenodo deposit: [10.5281/zenodo.15260640](https://doi.org/10.5281/zenodo.15260640).

## b) Supplemental Materials

### SM - 2025-05-27

- The replication package is now split into two separate notebooks: one dedicated to the datasets, and the other to the generation of figures used in version 2.0 of the preprint.  
- This ReleaseNote notebook has been added.  
- All in/out-degree histograms (computed per year, per month, and across all networks) have been updated to include `year == 3000`, which corresponds to the "All years" aggregate statistics.

---

### SM - 2025-05-16

The first complete version is now available, including:
- the QuickStart guide,  
- the replication packages (datasets and figures),  
- and supplemental materials SM01 to SM12.

---

### SM - 2025-04-28

First public release of the supplemental materials, including the QuickStart guide, and the initial version of the replication package (datasets).

The Python scripts are provided under the `local_utils` directory; however, they are not in their final form and should be considered an alpha release.

The datasets of the networks used in the study are available in a separate Zenodo deposit  
(DOI: [10.5281/zenodo.15260640](https://doi.org/10.5281/zenodo.15260640), $\sim50$ GB).  
This includes:
- the main dataset $O/RV/RL-O/RV/RL$ (over 2 billion nodes and $\sim4$ billion edges),  
- and two derived $O-(RV/RL)-O$ graphs ($\sim150$ million nodes and edges each).